src/sentry/models/dashboard.py

```diff
@@ -77,6 +78,32 @@ def get_prebuilt(organization, user, dashboard_id):
             return all_prebuilt_dashboards[dashboard_id]
         return None
 
+    @classmethod
+    def incremental_title(cls, organization, name):
+        """
+        Given a dashboard name that migh already exist, returns a new unique name that does not exist, by appending the word "copy" and an integer if necessary.
+        """
+
+        base_name = re.sub(r" ?copy ?(\d+)?$", "", name)
+        matching_dashboards = cls.objects.filter(
+            organization=organization, title__regex=rf"^{re.escape(base_name)} ?(copy)? ?(\d+)?$"
+        )
```

Comment:
Should the regex be _not_ optionally capturing the spaces, copy, and number? Since we know the intended format of the name we want to set, we can just query for ones that would potentially conflict (i.e. don't have to sift through the ones that won't conflict because copy and the number are optional)

```suggestion
        matching_dashboards = cls.objects.filter(
            organization=organization, title__regex=rf"^{re.escape(base_name)} copy \d+$"
        )
```

Reply:
Also might be a tiny optimization, but you can slap a `.values_list("title")` in there to just get back an iterable of titles, but this is a low priority suggestion considering the number of dashboards there are

Reply:
🤔 with a desired name "My Widgets", names "My Widgets", "My Widgets copy", and "My Widgets copy 1" are all conflicts, so I think I have to query for all of those, no? That's why all the spaces, "copy" and integer are all optional. Is there a way to condense that?

Added the `.values`, thanks for that! I'd rather have the dict than the tuple, I think

Reply:
Ah good point, I was trying to avoid cases like "My Widgets 1" but it's such a small set of cases to encounter. 

I also forgot to mention but you can add `flat_list=True` to `values_list` to get rid of the tuples